File: array/missing_ranges/missing_ranges.js
function missingRanges(arr, lo, hi) {
    const hashed = new Set(arr)
    for (let n = lo; n < hi; n++) {
        if (!hashed.has(n)) {
            console.log(n)
        }
    }
}

const inpt = [10, 12, 11, 15]
const low = 10
const hi = 15
console.log("input: ", inpt)
console.log("result: ")
missingRanges(inpt, low, hi)

File: array/missing_ranges/missing_ranges.go
package main

import "fmt"

func missingRanges(arr []int, low int, hi int) {
	hashed := make(map[int]bool)
	for _, i := range arr {
		hashed[i] = true
	}
	for n := low; n < hi; n++ {
		if _, ok := hashed[n]; !ok {
			fmt.Print(n)
			fmt.Print(" ")
		}
	}
}

func main() {
	inpt := []int{10, 12, 11, 15}
	low, hi := 10, 15
	fmt.Println("input: ", inpt)
	fmt.Print("result: ")
	missingRanges(inpt, low, hi)
}
